server/odc-service/src/main/java/com/oceanbase/odc/service/connection/ConnectionService.java

```diff
@@ -420,12 +420,32 @@ public List<ConnectionConfig> listByOrganizationIdIn(@NonNull Collection<Long> o
             resourceType = "ODC_PROJECT", indexOfIdParam = 0)
     public PaginatedData<ConnectionConfig> listByProjectId(@NotNull Long projectId, @NotNull Boolean basic) {
         List<ConnectionConfig> connections;
+        List<Long> environmentIds = repository.findByDatabaseProjectId(projectId).stream()
```

Comment:
There is no need to execute the same query `repository.findByDatabaseProjectId` twice on the line 423 and line 431.

Reply:
wow, very detailed, resolved.